### PR TITLE
Update to scala 2.11.8, plus some other deps and scalatest

### PR DESCRIPTION
**Pull request by vanvlack**
- update to support scala 2.11.8
 - Upgrade multiple other deps to newer versions
 - Drop simplespec for scalatest
 - Split up tests into separate classes vs nested

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -6,16 +6,32 @@
 
     <groupId>com.simple</groupId>
     <artifactId>jdub_2.11</artifactId>
-    <version>1.3.4</version>
+    <version>1.3.5</version>
```
Comment (vanvlack) [inline pom.xml]: Only question, should this be `1.4.0` vs `1.3.5`?

Comment (jarreds) [inline pom.xml]: I think a minor rev to `1.4` is warranted with all the dependency changes.

File: pom.xml [hunk 2]
```diff
@@ -6,16 +6,32 @@
...
     <name>Jdub for Scala ${scala.version}</name>
     <url>https://github.com/SimpleFinance/jdub</url>
     <description>Jdub is a Scala wrapper over JDBC.</description>
 
     <properties>
+        <grizzled-slf4j.version>1.2.0</grizzled-slf4j.version>
         <hikaricp.version>2.4.6</hikaricp.version>
```
Comment (jarreds) [inline pom.xml]: Let's roll to the HikariCP `2.4.9` release.

File: pom.xml [hunk 3]
```diff
@@ -157,7 +180,7 @@
             <plugin>
               <groupId>net.alchim31.maven</groupId>
```
Comment (jarreds) [inline pom.xml]: Nitpick. Indentation seems a bit off on these lines.

File: pom.xml (hunk 2 above)
Comment (vanvlack) [inline pom.xml]: Done

**Pull request merged**